server/src/main/scala/com/stratio/crossdata/server/CrossdataHttpServer.scala

```diff
@@ -53,11 +52,13 @@ class CrossdataHttpServer(config: Config, serverActor: ActorRef, implicit val sy
       entity(as[Multipart.FormData]) { formData =>
         // collect all parts of the multipart as it arrives into a map
         var path = ""
+        var file: java.io.File = null
+        //        var file:java.io.File
         val allPartsF: Future[Map[String, Any]] = formData.parts.mapAsync[(String, Any)](1) {
 
           case part: BodyPart if part.name == "fileChunk" =>
             // stream into a file as the chunks of it arrives and return a future file to where it got stored
-            val file = new java.io.File(s"/tmp/${part.filename.getOrElse("uploadFile")}")
+            file = new java.io.File(s"/tmp/${part.filename.getOrElse("uploadFile")}")
             path = file.getAbsolutePath
```

Comment:
If you have more than one part, this will not cause an inconsistency because you are sharing the same var across all the parts???